### DISCUSSION TITLE
How to disable identity authentication in Elsa 3.0？

### DISCUSSION THREAD

**Discussion by RingoAlanLee** [elsa-workflows/elsa-core#4016] [Q&A]
I hava a  web application(MVC) of .net6 and sign in with Azure AD
After set elsa server and design,I want to disable identity auth of elsa design.
What am I supposed to to do?I can't find anything about it in the documentation
![image](https://github.com/elsa-workflows/elsa-core/assets/97659120/29480293-e002-4284-bd4d-21c8e59c1346)
@sfmskywalker

**Comment by sfmskywalker** [comment]
You should be able to simply not install the identity feature. But then you do need to create a designer plugin that can authenticate outgoing HTTP requests made from the designer to the ASP.NET application. I imagine Azure AD stored some cookie containing some access token that your designer plugin would have to read.

This is a common scenario, so I think we should offer a designer plugin for Azure AD, in addition to others like Auth0.

**Reply by vnmnm** [reply]
@sfmskywalker For the API endpoints, I was able to disable by setting `EndpointSecurityOptions.DisableSecurity();` and then configuring our own auth provider to validate the incoming auth tokens. However, on the UI, how do we disable the login screen? I followed this [link ](https://v3.elsaworkflows.io/docs/installation/aspnet-apps-workflow-designer) to host the designer and don't see any option to not install the identity feature. There's only one package `Elsa.Workflows.Designer` that's getting installed and I don't see anything explicitly related to identity so I can exclude that.

**Reply by sfmskywalker** [reply]
The UI consists of modules that contain plugins. One of these plugins is called "login", which installs the login screen and intercepts outgoing HTTP requests to attach a bearer token.

You can opt-out of this plugin by writing a bit of JavaScript, like this:

```html
<body>
   <elsa-studio serverUrl={serverUrl} />
</body>

<script type="module">
  const studioElement = document.querySelector('elsa-studio');

  studioElement.addEventListener('initializing', e => {
    const pluginRegistry = e.detail.pluginRegistry;

    // Replace plugins with different ones, e.g. Auth0.
    pluginRegistry.replace('login', Auth0Plugin);

    // Or remove them altogether:
    pluginRegistry.remove('login');
  });
</script>
```

**Reply by ILAOHE** [reply]
<img width="1883" alt="image" src="https://github.com/elsa-workflows/elsa-core/assets/20391853/5183f3d8-02f5-4ffd-896b-d0d0ba294ac6">
According to the above operation, the design page of Elsa does not display properly, and the version number and new button are not realistic

**Reply by lemac1** [reply]
Did you managed to fix this? I have the same issue.
